blog/views.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.paginator import Paginator
+from django.shortcuts import redirect
+from django.views import generic
+
+from blog.models import Post, Commentary
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+
+class PostListView(generic.ListView):
+    model = Post
+    template_name = "blog/index.html"
+    paginate_by = 5
+    context_object_name = "post_list"
+
+
+@login_required
+def add_comment(request, pk):
+    if request.method == "POST":
+        comment_content = request.POST.get("comment")
+        if comment_content:
+            post = Post.objects.get(id=pk)
+            Commentary.objects.create(
+                post=post,
+                user=request.user,
+                content=comment_content
+            )
+            messages.success(request, "Comment successfully added!")
+        else:
+            messages.warning(request, "A comment cannot be empty!")
+    return redirect("blog:post-detail", pk=pk)
```

Comment:
The `add_comment` view is not idempotent. It changes the state of the system (by adding a new comment) even if it's called with the same parameters. According to the HTTP/1.1 spec, POST requests are not required to be idempotent. However, it's a good practice to make your views idempotent whenever possible. In this case, you could make the view idempotent by checking if a comment with the same content and user already exists for the post before creating a new one.